filament/backend/test/ImageExpectations.cpp

```diff
@@ -90,7 +92,15 @@ void ImageExpectation::evaluate() {
 }
 
 void ImageExpectation::compareImage() const {
+    std::cout << std::this_thread::get_id() << std::endl;
+    int attempts = 10;
+    while (attempts > 0 && !mResult.bytesFilled()) {
```

Comment:
It seems like backend testing should be deterministic. Why do we need to try this multiple times? Once we call `driver.finish()`, we should be guaranteed that all callbacks have completed.

Reply:
This snuck in from when I was diagnosing why EXPECT_IMAGE failed but readPixelsAndAssertHash didn't.
I thought the callback provided to the backend that sets `mResult.bytesFilled()` was running on another thread and it was a race condition, this was a quick alternative to setting up actual signals.
However those callbacks aren't made on another thread, so this is now removed.